test/MassMigs.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { Usage } from "../ts/interfaces";
+import { deployAll } from "../ts/deploy";
+import { TESTING_PARAMS } from "../ts/constants";
+import { ethers } from "@nomiclabs/buidler";
+import { StateTree } from "../ts/stateTree";
+import { AccountRegistry } from "../ts/accountTree";
+import { Account } from "../ts/stateAccount";
+import { TxMassMig } from "../ts/tx";
+import * as mcl from "../ts/mcl";
+import { Tree, Hasher } from "../ts/tree";
+import { allContracts } from "../ts/allContractsInterfaces";
+import { assert } from "chai";
+import { parseEvents } from "../ts/utils";
+
+const DOMAIN =
+    "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+
+describe("Rollup", async function() {
+    let Alice: Account;
+    let Bob: Account;
+    let contracts: allContracts;
+    let stateTree: StateTree;
+    let registry: AccountRegistry;
+    before(async function() {
+        await mcl.init();
+        mcl.setDomainHex(DOMAIN);
+    });
+
+    beforeEach(async function() {
+        const accounts = await ethers.getSigners();
+        contracts = await deployAll(accounts[0], TESTING_PARAMS);
+        stateTree = new StateTree(TESTING_PARAMS.MAX_DEPTH);
+        const registryContract = contracts.blsAccountRegistry;
+        registry = await AccountRegistry.new(registryContract);
+        const tokenID = 1;
+
+        Alice = Account.new(-1, tokenID, 10, 0);
+        Alice.setStateID(2);
+        Alice.newKeyPair();
+        Alice.accountID = await registry.register(Alice.encodePubkey());
+
+        Bob = Account.new(-1, tokenID, 10, 0);
+        Bob.setStateID(3);
+        Bob.newKeyPair();
+        Bob.accountID = await registry.register(Bob.encodePubkey());
+
+        stateTree.createAccount(Alice);
+        stateTree.createAccount(Bob);
+    });
+
+    it("submit a batch and dispute", async function() {
+        const tx = new TxMassMig(Alice.stateID, 0, 5, 1, 1, Alice.nonce + 1);
+        const signature = Alice.sign(tx);
+        const rollup = contracts.rollup;
+        const rollupUtils = contracts.rollupUtils;
+        var stateRoot = stateTree.root;
```

Comment:
We should use `const` whenever we can and use `let` whenever the variable would be reassigned.
We should never use `var`

```suggestion
        const stateRoot = stateTree.root;
```